### ISSUE TITLE
Tile Stamps and/or Random Mode usable with Object Layers

### ISSUE THREAD

**Issue by EdenIndustries** [bjorn/tiled#1147]
Currently it seems like Tile Stamps and Random Mode are both limited to Tile Layers. It would be great if they worked for Object Layers too!

**Comment by bjorn** [comment]
Do you mean supporting tiny tile layers as objects, or do you mean it would be nice to have a similar library of objects that are easy to share and access? The first is not really something I'd like to do, but the latter is definitely planned for the future.

**Comment by EdenIndustries** [comment]
Oh actually what I meant was essentially if you have a Tile Stamp selected and an Object Layer selected, you can use the Insert Tile functionality to place a tile chosen at random from the stamp into that Object Layer

**Comment by bjorn** [comment]
Ok, the difficulty with that is that the tile stamps can span multiple tiles, and tile objects can't. So in that case the result would generally not be what people may expect.

That said, supporting the random mode also for the insert tile tool would indeed be nice.

**Comment by EdenIndustries** [comment]
In the case where you have a stamp spanning multiple tiles, could you just loop through each tile in the stamp and place each one as an object?

**Comment by bjorn** [comment]
I could do that, but I don't want to do that... It could easily lead to people placing huge arrays of tile objects, which will not perform well and will be hard to edit.

**Comment by EdenIndustries** [comment]
Fair enough. If possible, do you mind giving me a quick notion of where to begin looking so I can implement it in my local copy? We use tile objects extensively so it'll be a massive time saver for us

**Comment by bjorn** [comment]
In `mainwindow.cpp`, you can find the following lines:
```c++
    connect(mTilesetDock, &TilesetDock::currentTileChanged,
            tileObjectsTool, &CreateObjectTool::setTile);
```
These tell the "Insert Tile" tool when the current tile in the tileset dock changes, so that it will it for inserting tiles.

In contrast, the "Stamp Brush" is told about the current stamp in `MainWindow::setStamp`. You can remove the connection above and instead tell the "Insert Tile" tool about the current stamp as done for the "Stamp Brush" and "Bucket Fill Tool".

For the rest, you'd need to look at adapting the `CreateTileObjectTool` to work based on the stamp instead of just the tile. You can look at `StampBrush` to see how it uses this `TileStamp` object.

**Comment by EdenIndustries** [comment]
Thanks, much appreciated!

**Comment by EdenIndustries** [comment]
Ok got this working, at least enough for our purposes. Thanks for your help pointing me in the right direction!

**Comment by bjorn** [comment]
You're welcome!

**Comment by eishiya** [comment]
-1 to placing multiple Tile Objects at once, because it'll encourage poor practices. #2863 (Tile Atlases) should help in the most common scenario where people usually want multi-tile objects, by making it easier to define a single tile from a larger region of the tileset.

Big +1 to placing a random Tile Object if multiple tiles are in the brush. I think a random tile should be picked and previewed every time you change tile location and not more frequently than that, so that you can have an opportunity to precisely place the object (unless you're snapping to the grid, in which case, this'll just make Insert Tile behave like the Stamp Brush in random mode).
